include/vcpkg/metrics.h

```diff
@@ -8,6 +8,68 @@
 
 namespace vcpkg
 {
+    template<typename T>
+    struct MetricEntry
+    {
+        T metric;
+        StringLiteral name;
+    };
+
+    enum class DefineMetric
+    {
+        AssetSource,
+        BinaryCachingAws,
+        BinaryCachingAzBlob,
+        BinaryCachingCos,
+        BinaryCachingDefault,
+        BinaryCachingFiles,
+        BinaryCachingGcs,
+        BinaryCachingHttp,
+        BinaryCachingNuget,
+        BinaryCachingSource,
+        ErrorVersioningDisabled,
+        ErrorVersioningNoBaseline,
+        GitHubRepository,
+        ManifestBaseline,
+        ManifestOverrides,
+        ManifestVersionConstraint,
+        RegistriesErrorCouldNotFindBaseline,
+        RegistriesErrorNoVersionsAtCommit,
+        VcpkgBinarySources,
+        VcpkgDefaultBinaryCache,
+        VcpkgNugetRepository,
+        VersioningErrorBaseline,
+        VersioningErrorVersion,
+        X_VcpkgRegistriesCache,
+        X_WriteNugetPackagesConfig,
+        _COUNT // always keep COUNT last
```

Comment:
This breaks `lex.name`; it can't be `_Ugly`.

Reply:
```suggestion
        COUNT // always keep COUNT last
```